content/docs/internals/architecture.md

```diff
@@ -11,38 +11,46 @@ Pomerium sits between end users and services requiring strong authentication. Af
 
 ![pomerium architecture diagram](./img/architecture/pomerium-system-context.svg)
 
-## Component Level
+## Component level
 
-Pomerium is composed of 4 logical components:
+Pomerium is composed of four logical components:
 
-- Proxy Service ([What is a proxy?](https://www.pomerium.com/blog/proxy-vs-reverse-proxy/))
+### Proxy service
+The Proxy service receives requests from the client and routes requests between the client, upstream services, and other Pomerium components. 
 
-  - All user traffic flows through the proxy
-  - Verifies all requests with Authentication service
-  - Directs users to Authentication service to establish session identity
-  - Processes policy to determine external/internal route mappings
-  - Receives user OIDC tokens and session data from Authentication service and stores in Databroker service
+The following steps outline how the Proxy service handles a request at a high level:
 
-- Authentication Service
+1. The client makes a request to access the target resource 
+1. The Proxy service receives the request and redirects it to the Authorization service before proxying it
+1. If the Authorization service doesn’t see a session cookie, the Proxy service redirects the request to the Authentication service to verify the client’s identity
+1. After the Authentication service verifies the client’s identity and saves a local session cookie, the Proxy service redirects the session data to the Databroker service
+1. Before redirecting the client to the target resource, the Proxy service checks permissions with the Authorization service, then maps a route based on the internal and external routes defined in the route’s policy
 
-  - Handles authentication flow to your IdP as needed
-  - Handles identity verification after initial Authentication
-  - Establishes user session cookie
-  - Stores user OIDC tokens in Proxy service
+### Authentication service
+The Authentication service is responsible for authenticating users against an Identity Provider (IdP) and establishing user sessions. By incorporating OAuth 2.0 and OIDC protocols into the authentication flow, the Authentication service provides single sign-on authentication that enables it to extract user identity details and session data necessary for managing Pomerium sessions. 
 
-- Authorization Service
+At a high level, when the Authentication service first receives a request from the Proxy service, it:
+1. Sends an HTTP request on behalf of the client to the IdP
+1. Completes the authentication flow with the IdP and extracts relevant session data, such as device credentials, user ID, issuance and expiration times, OAuth tokens, and any OIDC claims (scopes) provided by the IdP
+1. Saves session data to a local session cookie and redirects the client with session data encrypted in URL parameters to the Proxy service
+1. Verifies user identity after the session expires
```

Comment:
After a session expires its as if the user is no longer logged in. Authenticate merely runs the login process again. It doesn't verify the user identity.